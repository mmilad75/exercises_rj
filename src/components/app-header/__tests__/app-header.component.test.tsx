import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import AppHeader from "../app-header.component";

describe("AppHeader component", () => {
    it("should render correctly", () => {
        const { asFragment } = render(
            <AppHeader selected="counter" onSelect={jest.fn()} />
        );
        expect(asFragment()).toMatchSnapshot();
    });

    it("should display all items", () => {
        render(<AppHeader selected="counter" onSelect={jest.fn()} />);

        const items = screen.getAllByRole("button");

        expect(items.length).toEqual(3);
    });

    it("should call onSelect when clicking on items", () => {
        const mockOnSelect = jest.fn();
        render(<AppHeader selected="counter" onSelect={mockOnSelect} />);

        const items = screen.getAllByRole("button");

        userEvent.click(items[0]);

        expect(mockOnSelect).toBeCalled();
    });
});
