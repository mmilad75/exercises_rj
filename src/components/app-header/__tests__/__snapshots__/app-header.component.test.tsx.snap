// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppHeader component should render correctly 1`] = `
<DocumentFragment>
  <header
    role="navigation"
    style="display: flex; justify-content: center; background: rgb(226, 226, 226); margin-bottom: 20px; border-radius: 8px;"
  >
    <div
      role="button"
      style="margin-right: 16px; padding: 20px; cursor: pointer;"
    >
      Users
    </div>
    <div
      role="button"
      style="margin-right: 16px; padding: 20px; cursor: pointer; color: rgb(29, 161, 242);"
    >
      Counter
    </div>
    <div
      role="button"
      style="margin-right: 16px; padding: 20px; cursor: pointer;"
    >
      Login
    </div>
  </header>
</DocumentFragment>
`;
